tools/update_unicode_iso8859_tables.dart

```diff
@@ -0,0 +1,300 @@
+}
+
+/// Shared HTTP client used by [fetchTable], initialized if needed.
+///
+/// Closed by [generateTables] when done fetching.
+HttpClient? httpClient;
+
+/// Fetches table file content, parses and builds constant string declarations.
+Future<String> generateTables() async {
+  // Used for caching fetched files.
+  // Useful if running the script multiple times.
+  var cacheDir = Directory(cachePath);
+  if (!cacheDir.existsSync()) cacheDir.createSync(recursive: true);
+
+  var buffer = StringBuffer();
+  // For each ISO-8859-n, n > 1, fetch the table,
+  // and generate a constant string corresponding to the mapping
+  // into `buffer`.
+  // There is no ISO-8859-12. Don't include ISO-8859-1,
+  // since it's already included in `dart:convert`.
+  for (var i = 2; i <= 16; i++) {
+    if (i == 12) continue; // No ISO-8859-12.
+    await generateTableFor(buffer, i);
+  }
+
+  httpClient?.close();
+  return buffer.toString();
+}
+
+/// Generates the constant string for a single ISO-8859 code page.
+Future<void> generateTableFor(StringSink buffer, int isoNumber) async {
+  var (uri, content) = await fetchTable(isoNumber);
+  var text = parseTable(isoNumber, content);
+  buffer
+    ..writeln()
+    ..writeln('// Characters of ISO-8859-$isoNumber.')
+    ..write('const iso8859_$isoNumber = // From $uri');
+  // Write 8 chars per line, as Unicode escapes.
+  for (var i = 0; i < text.length;) {
+    var charsOnLine = nextRangeLength(text, i, 74);
+    buffer.write('\n    \'');
+    for (var c = 0; c < charsOnLine; c++) {
+      writeChar(buffer, text.codeUnitAt(i + c));
+    }
+    i += charsOnLine;
+    buffer.write('\'');
+  }
+  buffer.writeln(';');
+}
+
+// Code units that are encoded specially in the generated literals.
+const specialChars = {
+  0x08: r'\b',
+  0x09: r'\t',
+  0x0a: r'\n',
+  0x0b: r'\v',
+  0x0c: r'\f',
+  0x0d: r'\r',
+  0x24: r'\$',
+  0x27: r"\'",
+  0x7f: r'\x7F',
+  0x5c: r'\\',
+};
+
+/// Write a single code unit into a string literal.
+///
+/// Writes all code units above 0xFF as `\uHHHH`.
+/// Writes all printable ASCII codes as themselves,
+/// possibly escaped.
+/// Writes characters that have escapes, like `\n`, as those.
+/// The remaining code units in the 0x00..0xFF range are written as `\xHH`
+void writeChar(StringSink buffer, int charCode) {
+  if (specialChars[charCode] case var s?) {
+    buffer.write(s);
+  } else if (charCode >= 0x20 && charCode < 0x7f) {
+    buffer.writeCharCode(charCode);
+  } else if (charCode <= 0xFF) {
+    buffer
+      ..write(charCode < 0x10 ? r'\x0' : r'\x')
+      ..write(charCode.toRadixString(16).toUpperCase());
+  } else {
+    buffer
+      ..write(r'\u')
+      ..write(charCode.toRadixString(16).toUpperCase().padLeft(4, '0'));
+  }
+}
+
+/// Find the number of characters that fits in a single-line string literal.
+///
+/// Finds the largest number of characters starting from [from],
+/// which is a power of 2, and where the characters representation,
```

Comment:
Why do you want to limit this to powers-of-2?

Reply:
Mostly for a miniscule amount of readability. Which likely isn't really needed.

I started just doing eight `\uHHHH` chars per line, which works, and ensures that you *can* visually find the individual characters in the mapping if you really want to.
But it's so big, especially for the ASCII part, so I tried to reduce the size by using `\xHH` encodings and visible ASCII chars, but it's entirely a source-size-and-readability thing, it shouldn't make any difference to the string.

But, to make it still somewhat possible to eyeball what character is at which position, I wanted the lines to all be
power-of-2 aligned.
Or really, I wanted them to have lengths that are powers of 2 *and* be aligned by their size.
I think I might have gotten the latter, just because of how things end up being encoded.

Reply:
If we don't care about the readability of the generated file, you can do this however you want.

Reply:
Yeah. I'm coming around to that too. 
In which case anything other than repeated `\uHHHH` entries is a waste of developer time and complexity budget. (I'll still put 8 of them on each line, to get a little alignment. I like alignment. Maybe even has room for a `// 0xc0..0xc7` comment. And there I go trying for readability again. Because debugging ...? Don't debug generated code, debug the generator!)